src/core/prompts/instructions/generate-rules.ts

```diff
@@ -0,0 +1,125 @@
+export interface RuleInstruction {
+	path: string
+	focus: string
+	analysisSteps: string[]
+}
+
+export interface RulesGenerationOptions {
+	selectedRuleTypes: string[]
+	addToGitignore: boolean
+	alwaysAllowWriteProtected: boolean
+	includeCustomRules: boolean
+	customRulesText: string
+}
+
+export function generateRulesInstructions(
+	ruleInstructions: RuleInstruction[],
+	options: RulesGenerationOptions,
+): string {
+	const { addToGitignore, alwaysAllowWriteProtected, includeCustomRules, customRulesText } = options
+
+	return `Analyze this codebase and generate comprehensive rules for AI agents working in this repository.
+
+Your task is to:
+
+1. **Analyze the project structure** by:
+${ruleInstructions.map((rule) => `   - For ${rule.path.split("/").pop()}: ${rule.analysisSteps.join("; ")}`).join("\n")}
+
+2. **Look for existing rule files** that might provide guidance:
+   - Check for CLAUDE.md, .cursorrules, .cursor/rules, or .github/copilot-instructions.md
+   - If found, incorporate and improve upon their content
+
+3. **Generate and save the following rule files**:
+${ruleInstructions
+	.map(
+		(rule, index) => `
+   ${index + 1}. **${rule.path}**
+      - Focus: ${rule.focus}${alwaysAllowWriteProtected ? "\n      - The directory has already been created for you" : "\n      - Create the necessary directories if they don't exist"}
+      - Always overwrite the existing file if it exists
+      - Use the \`write_to_file\` tool to save the content${alwaysAllowWriteProtected ? "\n      - Note: Auto-approval for protected file writes is enabled, so you can write to .roo directories without manual approval" : "\n      - Note: You will need to approve the creation of protected directories and files"}`,
```

Comment:
Does the prompt need to know whether alwaysAllowWriteProtected is enabled? I wouldn't think so.